packages/yourloops/lib/selected-team/selected-team.hook.tsx

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2022, Diabeloop
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { SelectedTeamContextResult } from './selected-team-context.model'
+import { useState } from 'react'
+import { Team, useTeam } from '../team'
+import TeamUtils from '../team/team.util'
+
+const LOCAL_STORAGE_SELECTED_TEAM_ID_KEY = 'selectedTeamId'
+
+export function useSelectedTeamProviderCustomHook(): SelectedTeamContextResult {
+  const { getMedicalAndPrivateTeams } = useTeam()
+
+  const getSortedTeams = (): Team[] => {
+    const teams = getMedicalAndPrivateTeams()
+    return TeamUtils.sortTeams(teams)
+  }
+
+  const sortedTeams = getSortedTeams()
```

Comment:
```suggestion
  const getSortedTeams = useCallback((): Team[] => {
    const teams = getMedicalAndPrivateTeams()
    return TeamUtils.sortTeams(teams)
  }, [getMedicalAndPrivateTeams])

  const sortedTeams = useMemo(() => {
    return getSortedTeams()
  }, [getSortedTeams])
```

Some opti, this does not need to be recomputed on every `setSelectedTeamId` call.

Another great and even more optimal solution would be to move this code directly into `getDefaultTeamId`. What do you think?

Reply:
I think I did what you asked (including the `getDefaultTeamId` changes), is this what you had in mind?

Reply:
Yes it is better! But with this solution `getSortedTeams` does not need to be in a useCallback. Sorry, I did not make that clear in my comment :slightly_frowning_face: